UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -26,19 +26,55 @@ public class MetaTileMap : MonoBehaviour
 
 		private Stopwatch stopwatch = new Stopwatch();
 
-		private readonly Dictionary<Layer, Dictionary<Vector3Int, TileLocation>> PresentTiles =
-			new Dictionary<Layer, Dictionary<Vector3Int, TileLocation>>();
-
-		public Dictionary<Layer, Dictionary<Vector3Int, TileLocation>> PresentTilesNeedsLock => PresentTiles;
-
-		private readonly Dictionary<Layer, Dictionary<Vector3Int, List<TileLocation>>> MultilayerPresentTiles =
-			new Dictionary<Layer, Dictionary<Vector3Int, List<TileLocation>>>();
-
-		public Dictionary<Layer, Dictionary<Vector3Int, List<TileLocation>>> MultilayerPresentTilesNeedsLock =>
+		private readonly ChunkedTileMap<TileLocation>[] PresentTiles =
+			new ChunkedTileMap<TileLocation>[]
+			{
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
+				new ChunkedTileMap<TileLocation>(),
```

Comment:
why

Reply:
doesn't want to pre-populate for some reason